### PR TITLE
Adding extra debug logs

### PR DESCRIPTION
**Pull request by thesattiraju**
**What this PR does / why we need it**: A possible solution for #5941

### CODE REVIEW

**Review by bansalaseem — APPROVED** [review]

File: cmd/helm/init.go [hunk 1]
```diff
@@ -284,6 +284,9 @@ func (i *initCmd) run() error {
 				}
 				fmt.Fprintln(i.out, "\nTiller (the Helm server-side component) has been upgraded to the current version.")
 			} else {
+				if settings.Debug {
+					fmt.Fprintln(i.out, fmt.Sprintf("[INFO] The error received while trying to init: %s", err))
+				}
```
Comment (mattfarina) [inline cmd/helm/init.go]: I like the desire to display the debug information. But, can you use the `debug` function to do this instead? It's part of the `main` package here in `printer.go`. It has the if statement wrapper, etc.

**Review by mattfarina — CHANGES_REQUESTED** [review]

File: cmd/helm/init.go (hunk 1 above)
Comment (ds-ms) [reply cmd/helm/init.go]: @mattfarina Updated the logic to use `debug` function instead

File: cmd/helm/init.go [hunk 2]
```diff
@@ -284,6 +284,7 @@ func (i *initCmd) run() error {
 				}
 				fmt.Fprintln(i.out, "\nTiller (the Helm server-side component) has been upgraded to the current version.")
 			} else {
+				debug("[INFO] The error received while trying to init: %s", err)
```
Comment (hickeyma) [inline cmd/helm/init.go]: Remove the '[INFO]' label as debug function uses '[debug]' label.

**Review by hickeyma — CHANGES_REQUESTED** [review]

File: cmd/helm/init.go (hunk 2 above)
Comment (ds-ms) [reply cmd/helm/init.go]: Updated

**Review by hickeyma — APPROVED** [review]
LGTM, thanks.

**Review by mattfarina — APPROVED** [review]

**Comment by thomastaylor312** [comment]
Init is being removed in v3, so marking as port complete

**Pull request merged**